src/Core/src/Platform/iOS/Swizzle.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Foundation;
+using ObjCRuntime;
+
+namespace Microsoft.Maui.Platform.iOS
+{
+	internal class Swizzle<TDelegate> : IDisposable where TDelegate : class
+	{
+		protected IntPtr originalMethod;
+		protected IntPtr originalImpl;
+		protected IntPtr swizzleSel;
+		protected IntPtr newImpl;
+		protected bool isClassMethod;
+
+		protected TDelegate dlg; // Your delegate
+
+		public Swizzle(Class classToSwizzle, IntPtr selector, TDelegate del, bool isClassMethod = false)
+		{
+			dlg = del;
+			swizzleSel = selector;
+
+			originalMethod = isClassMethod ? LibObjc.class_getClassMethod(classToSwizzle.Handle, swizzleSel) : LibObjc.class_getInstanceMethod(classToSwizzle.Handle, swizzleSel);
```

Comment:
What if there's no original method (the type in question doesn't implement the selector)?